programs/keeper/clickhouse-keeper.cpp

```diff
@@ -1,6 +1,32 @@
+#include <Common/StringUtils/StringUtils.h>
+#include "config_tools.h"
+
+
 int mainEntryClickHouseKeeper(int argc, char ** argv);
 
+#if ENABLE_CLICKHOUSE_KEEPER_CLIENT
+int mainEntryClickHouseKeeperClient(int argc, char ** argv);
+#endif
+
 int main(int argc_, char ** argv_)
 {
+#if ENABLE_CLICKHOUSE_KEEPER_CLIENT
+
+    if (argc_ >= 2)
+    {
+        /// 'clickhouse-keeper --client ...' and 'clickhouse-keeper client ...' are OK
+        if (strcmp(argv_[1], "--client") == 0 || strcmp(argv_[1], "client") == 0)
+        {
+            if (--argc_ > 1)
+                std::memmove(&argv_[1], &argv_[2], sizeof(char *) * (argc_ - 1));
+
+            return mainEntryClickHouseKeeperClient(argc_, argv_);
```

Comment:
no need to move anything, why not just send `argv_ + 1`?

Reply:
Because we want to remove `client` from `./clickhouse-keepr client [...]`. And we can't remove argv[0], because the application relies on it 
https://github.com/ClickHouse/ClickHouse/blob/master/base/poco/Util/src/Application.cpp#L343

Reply:
can you just move the first to second then?

Reply:
Yeah, that makes sense